src/content/BatchRequestContentCollection.ts

```diff
@@ -0,0 +1,140 @@
+  /**
+   * @private
+   * @static
+   * The list of requests to be sent in the batch request
+   */
+  public readonly batchRequests: BatchItem[] = [];
+
+  /**
+   * @public
+   * @constructor
+   * Creates the BatchRequestContentCollection instance
+   * @param {RequestAdapter} requestAdapter - The request adapter to be used for executing the requests
+   * @param {ErrorMappings} errorMappings - The error mappings to be used while deserializing the response
+   * @param batchLimit - The maximum number of requests that can be sent in a single batch request
+   */
+  constructor(requestAdapter: RequestAdapter, errorMappings: ErrorMappings, batchLimit: number) {
+    if (!requestAdapter) {
+      const error = new Error("Request adapter is undefined, Please provide a valid request adapter");
+      error.name = "Invalid Request Adapter Error";
+      throw error;
+    }
+    this.requestAdapter = requestAdapter;
+    if (!errorMappings) {
+      const error = new Error("Error mappings are undefined, Please provide a valid error mappings");
+      error.name = "Invalid Error Mappings Error";
+      throw error;
+    }
+    this.errorMappings = errorMappings;
+
+    if (!batchLimit) {
+      batchLimit = DEFAULT_BATCH_LIMIT;
+    }
+    this.batchLimit = batchLimit;
+  }
+
+  /**
+   * Executes the batch requests asynchronously.
+   *
+   * @returns {Promise<BatchResponseContent | undefined>} A promise that resolves to the batch response content or undefined.
+   * @throws {Error} If the batch limit is exceeded.
+   */
+  public async postAsync(): Promise<BatchResponseContent | undefined> {
+    // chuck the batch requests into smaller batches
+    const batches = this.chunkArray(this.batchRequests, 19);
+
+    if (batches.length > this.batchLimit) {
+      const error = new Error(`Batch limit exceeded, Please provide a batch limit of ${this.batchLimit} or less`);
+      error.name = "Batch Limit Exceeded Error";
+      throw error;
+    }
+
+    // loop over batches and create batch request body
+    const batchResponseBody: BatchResponseContent[] = [];
+    for (const batch of batches) {
+      const requestContent = new BatchRequestContent(this.requestAdapter, this.errorMappings);
+      requestContent.addRequests(batch);
+      const response = await requestContent.postAsync();
+      if (response) {
+        batchResponseBody.push(response);
+      }
+    }
+
+    const mergedResponses = batchResponseBody.flatMap(batchResponse =>
+      Array.from(batchResponse.getResponsesIterator(), ([, value]) => value),
+    );
+    return new BatchResponseContent({ responses: mergedResponses });
+  }
+
+  /**
+   * @public
+   * Receives a request information object, converts it and adds it to the batch request execution chain
+   * @param requestInformation - The request information object
+   * @param batchId - The batch id to be used for the request
+   */
+  public addBatchRequest(requestInformation: RequestInformation, batchId?: string): BatchItem {
```

Comment:
We're also missing the `addRequest` method to add a `BatchItem` directly. Similar to the other class